test/test-spawn.c

```diff
@@ -90,7 +90,15 @@ static void kill_cb(uv_process_t* process,
 #else
   ASSERT(exit_status == 0);
 #endif
+#if defined(__APPLE__)
+  /*
+   * At least starting with Darwin Kernel Version 16.4.0, sending a SIGTERM to a
+   * process that is still starting up kills it with SIGKILL instead of SIGTERM.
```

Comment:
Can you add a link to the issue for future reference?  Would save a bit of digging through the history to find it again.  Other than that LGTM.